fix: [jaspr_riverpod] vsyncOverride is missing a variable

**Description**

When trying to run the `examples/flutter_embedding` example, get the following error:

```sh
 ~/gadfly/temp/jaspr/examples/flutter_embedding/ [main*] jaspr serve
✓ [CLI] Starting jaspr in development mode... (5.3s)
[BUILDER] [ERROR] Error compiling dartdevc module:jaspr_riverpod|lib/jaspr_riverpod.ddc.js
[BUILDER] [ERROR] 
[BUILDER] [ERROR] packages/jaspr_riverpod/src/framework.dart:317:5: Error: The getter 'vsyncOverride' isn't defined for the class '_UncontrolledProviderScopeElement'.
[BUILDER] [ERROR] - '_UncontrolledProviderScopeElement' is from 'package:jaspr_riverpod/src/framework.dart' ('packages/jaspr_riverpod/src/framework.dart').
[BUILDER] [ERROR] Try correcting the name to the name of an existing getter, or defining a getter or field named 'vsyncOverride'.
[BUILDER] [ERROR] vsyncOverride ??= _jasprVsync;
[BUILDER] [ERROR] ^^^^^^^^^^^^^
[BUILDER] [ERROR] packages/jaspr_riverpod/src/framework.dart:317:5: Error: The setter 'vsyncOverride' isn't defined for the class '_UncontrolledProviderScopeElement'.
[BUILDER] [ERROR] - '_UncontrolledProviderScopeElement' is from 'package:jaspr_riverpod/src/framework.dart' ('packages/jaspr_riverpod/src/framework.dart').
[BUILDER] [ERROR] Try correcting the name to the name of an existing setter, or defining a setter or field named 'vsyncOverride'.
[BUILDER] [ERROR] vsyncOverride ??= _jasprVsync;
[BUILDER] [ERROR] ^^^^^^^^^^^^^
[BUILDER] [ERROR] packages/jaspr_riverpod/src/framework.dart:373:9: Error: The getter 'vsyncOverride' isn't defined for the class '_UncontrolledProviderScopeElement'.
[BUILDER] [ERROR] - '_UncontrolledProviderScopeElement' is from 'package:jaspr_riverpod/src/framework.dart' ('packages/jaspr_riverpod/src/framework.dart').
[BUILDER] [ERROR] Try correcting the name to the name of an existing getter, or defining a getter or field named 'vsyncOverride'.
[BUILDER] [ERROR] if (vsyncOverride == _jasprVsync) {
[BUILDER] [ERROR] ^^^^^^^^^^^^^
[BUILDER] [ERROR] packages/jaspr_riverpod/src/framework.dart:374:7: Error: The setter 'vsyncOverride' isn't defined for the class '_UncontrolledProviderScopeElement'.
[BUILDER] [ERROR] - '_UncontrolledProviderScopeElement' is from 'package:jaspr_riverpod/src/framework.dart' ('packages/jaspr_riverpod/src/framework.dart').
[BUILDER] [ERROR] Try correcting the name to the name of an existing setter, or defining a setter or field named 'vsyncOverride'.
[BUILDER] [ERROR] vsyncOverride = null;
```

**Steps To Reproduce**

<!-- Please include full steps to reproduce so that we can reproduce the problem. -->

1. Clone jaspr repo (this Issue was made against the commit: `3495cfd0693fed99a34ffec0f9f0b8ac9bbfb18e`)
2. `cd examples/flutter_embedding`
3. install dependencies
4. `jaspr serve`

**Doctor Output**

```sh
 ~/gadfly/temp/jaspr/examples/flutter_embedding/ [main*] jaspr doctor
[✓] Jaspr CLI (Version 0.9.3)
  • Dart Version 3.1.1 (stable) (Tue Sep 5 12:20:14 2023 +0000) on "macos_x64" at /Users/matthew/fvm/versions/3.13.3/bin/cache/dart-sdk/bin/dart
  • Running on macos Version 13.3.1 (a) (Build 22E772610a) - Locale en-US
  • Analytics: Enabled

[✓] Current Project
  • Dependencies on core packages:
    • jaspr: ^0.9.0
    • jaspr_builder: ^0.9.0 (dev)
    • jaspr_web_compilers: ^4.0.5 (dev)
    • jaspr_flutter_embed: ^0.1.4
    • jaspr_riverpod: ^0.3.6
  • Uses server-side rendering: true
  • Uses experimental compilers: true
  • Uses flutter embedding: true
```

**Expected Behavior**

For the flutter_embedding example to have been served to localhost:8080 without hitting an error.

**Additional Context**

I think this [commit](https://github.com/schultek/jaspr/commit/32e8b31f71b14cf9f4d63aaed5893584c2431086) may have introduced the regression.

--- a/AUTHORS
+++ b/AUTHORS
@@ -6,3 +6,4 @@
 Kilian Schulte (schultek) - Creator & Lead Developer
 Martin Jablečník (mjablecnik) - Contributor
 Jeff Ward (fuzzybinary) - Contributor
+Matthew Jaoudi (gadfly361) - Contributor

--- a/packages/jaspr_riverpod/CHANGELOG.md
+++ b/packages/jaspr_riverpod/CHANGELOG.md
@@ -1,4 +1,4 @@
-## Unreleased 0.3.8
+## Unreleased patch
 
 - Upgrade `riverpod` to `v2.4.6`
 - Update `jaspr_riverpod`'s dependency on `riverpod`'s `vsyncOverride` to `flutterVsyncs`

--- a/packages/jaspr_riverpod/CHANGELOG.md
+++ b/packages/jaspr_riverpod/CHANGELOG.md
@@ -1,3 +1,8 @@
+## Unreleased 0.3.8
+
+- Upgrade `riverpod` to `v2.4.6`
+- Update `jaspr_riverpod`'s dependency on `riverpod`'s `vsyncOverride` to `flutterVsyncs`
+
 ## 0.3.7
 
 - Fixed bug with SyncProvider.

--- a/packages/jaspr_riverpod/lib/src/framework.dart
+++ b/packages/jaspr_riverpod/lib/src/framework.dart
@@ -314,7 +314,7 @@ class _UncontrolledProviderScopeElement extends InheritedElement {
       debugCanModifyProviders ??= _debugCanModifyProviders;
     }
 
-    vsyncOverride ??= _jasprVsync;
+    flutterVsyncs.add(_jasprVsync);
     super.mount(parent, prevSibling);
   }
 
@@ -370,9 +370,7 @@ class _UncontrolledProviderScopeElement extends InheritedElement {
       debugCanModifyProviders = null;
     }
 
-    if (vsyncOverride == _jasprVsync) {
-      vsyncOverride = null;
-    }
+    flutterVsyncs.remove(_jasprVsync);
     super.unmount();
   }
 

--- a/packages/jaspr_riverpod/pubspec.yaml
+++ b/packages/jaspr_riverpod/pubspec.yaml
@@ -14,7 +14,7 @@ environment:
 dependencies:
   jaspr: ^0.9.0
   meta: ^1.9.1
-  riverpod: ^2.3.2
+  riverpod: ^2.4.6
 
 dev_dependencies:
   jaspr_test: '>=0.1.0 <1.0.0'

